src/plugins/kibana_react/public/core/context.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import * as React from 'react';
+import { KibanaReactContextValue, Core } from './types';
+
+export const context = React.createContext<KibanaReactContextValue>({
+  core: {},
```

Comment:
Yes, the current typings already should allow to achieve this, but I don't see how this would be useful. Why don't you want to provide the whole `core` but just part of it?

Reply:
The use case would be a large application where you want to clearly understand your integration points with core are... If all of core is just thrown into the context, it isn't immediately obvious which parts of core you are actually using in your application, because everything has access to everything.

But if you narrow core down to just the services you _know_ you need, then doing things like upgrades becomes easier in the future. If core introduces a breaking change, it becomes more obvious exactly where your integration points are, because you don't have to go digging through every component to determine which core services are being used via context.

Anyway... that discussion is a bit tangential to the PR, but certainly worth a follow up discussion. Mostly what I wanted to confirm was that this implementation wouldn't restrict downstream devs from doing something like this if they wanted to.